frontend_tests/node_tests/typeahead_helper.js

```diff
@@ -291,7 +291,7 @@ test("sort_languages", () => {
     test_langs = th.sort_languages(test_langs, "java");
     assert.deepEqual(test_langs, ["java", "javascript", "js", "j"]);
     test_langs = th.sort_languages(test_langs, "j");
-    assert.deepEqual(test_langs, ["j", "javascript", "java", "js"]);
+    assert.deepEqual(test_langs, ["j", "javascript", "js", "java"]);
```

Comment:
It seems not ideal to have two copies of `javascript` in the same place.

I think we might want to make the typeahead nicer, to have a single entry for each language, displaying and matching on all aliases, sorta like what you have in the playgrounds view.

Reply:
@timabbott I'm guessing we don't want to replace the aliases with canonical names, like the playgrounds typeheads, but continue to display the aliases? And instead of displaying all aliases, we only display a single entry - E.g: we would display `["j", "javascript", "java"]` in the above example. Choosing what the single entry would be might be tricky - we could perhaps use the alias we define in the `default` section of `lang.json`?

Reply:
@timabbott The latest comment I added implements this sort of behavior. Let me know if that's what you had in mind.